[MSE] Bring back some useful SourceBuffer logs

eocanha opened the pull request:
#### 445669890d9b15b6f2c452a7a03f2057c912ac8f
<pre>
[MSE] Bring back some useful SourceBuffer logs
<a href="https://bugs.webkit.org/show_bug.cgi?id=247122">https://bugs.webkit.org/show_bug.cgi?id=247122</a>

Follow up patch for <a href="https://commits.webkit.org/256224@main">https://commits.webkit.org/256224@main</a>, this time applying
Jer Noble&apos;s suggestions to use StringConcatenation instead of a printf-like
syntax.

Reviewed by Jer Noble.

* Source/WebCore/Modules/mediasource/SourceBuffer.cpp:
(WebCore::SourceBuffer::appendBufferInternal): Use string concatenation.
(WebCore::SourceBuffer::sourceBufferPrivateAppendComplete): Ditto.
* Source/WebCore/platform/graphics/PlatformTimeRanges.cpp:
(WebCore::PlatformTimeRanges::toString const): Added String conversion method.
* Source/WebCore/platform/graphics/PlatformTimeRanges.h:
(WTF::LogArgument&lt;PlatformTimeRanges&gt;::toString): Ditto.

Canonical link: <a href="https://commits.webkit.org/256317@main">https://commits.webkit.org/256317@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2923d81aa7179a5aecd9cec4abb0958c03893db4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/95286 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/4568 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/28346 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/104879 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/165144 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/99274 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/4565 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/33283 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/87652 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/100764 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/100950 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/3327 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/81981 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/30333 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/85228 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/87156 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/73232 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/39009 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/18711 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/36817 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/20005 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4367 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/40768 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/42713 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/42752 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/39243 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

jernoble commented on Source/WebCore/Modules/mediasource/SourceBuffer.cpp:
```diff
@@ -592,9 +590,7 @@ void SourceBuffer::sourceBufferPrivateAppendComplete(AppendResult result)
-    StringPrintStream message;
-    message.printf("SourceBuffer::sourceBufferPrivateAppendComplete(%p) - buffered = %s", this, toString(m_private->buffered()->ranges()).utf8().data());
-    DEBUG_LOG(LOGIDENTIFIER, message.toString());
+    DEBUG_LOG(LOGIDENTIFIER, "buffered = ", m_private->buffered()->ranges().toString());
```
This is fine as-is, but I _think_ it should be possible to drop the final `.toString()`, with the addition of a LogArgument template for PlatformTimeRanges.

eocanha replied:
It is, yes, but I was doing it wrong.

I tried, the build failed with an indecipherable template error and I just gave up and used `toString()`. After digging a bit more, it turns out that I [already had the `LogArgument<PlatformTimeRanges>::toString()` code in place](https://github.com/WebKit/WebKit/pull/6044/commits/e1ccf51b4a4700f889d6529041a27a5b14da9bfa#diff-0766b78fb2791875107e1415c7aeae8098e7dbca2862b9f4b68c7d6bb5ab794bR170), but defined in the WebCore namespace instead of the WTF one.

I've modified the definition in `PlatformTimeRanges.h` to be on WTF and now it works (see new patch). I was wondering why [it worked out of the box for MediaTime](https://github.com/WebKit/WebKit/blob/main/Source/WTF/wtf/MediaTime.h#L193) and not for PlatformTimeRanges until I realized that MediaTime belongs to WTF.

jernoble reviewed (commented):
r=me

eocanha commented on Source/WebCore/platform/graphics/PlatformTimeRanges.h:
```diff
@@ -165,4 +166,13 @@ std::optional<PlatformTimeRanges> PlatformTimeRanges::decode(Decoder& decoder)
 
 } // namespace WebCore
 
+namespace WTF {
```
@jernoble, I fear that I'm committing some kind of layer violation by declaring WTF stuff in a WebCore header. Can you please have a look at the new patch and approve the PR here in GitHub only if adding WTF extra declarations in PlatformTimeRanges is the right thing to do?

jernoble replied:
The layering violation would be declaring WebCore things in a WTF header. :) 

What you're doing here is completely fine, and is a common pattern in other areas of the WebKit codebase.

eocanha committed:
[MSE] Bring back some useful SourceBuffer logs
https://bugs.webkit.org/show_bug.cgi?id=247122

Follow up patch for https://commits.webkit.org/256224@main, this time applying
Jer Noble's suggestions to use StringConcatenation instead of a printf-like
syntax.

Reviewed by Jer Noble.

* Source/WebCore/Modules/mediasource/SourceBuffer.cpp:
(WebCore::SourceBuffer::appendBufferInternal): Use string concatenation.
(WebCore::SourceBuffer::sourceBufferPrivateAppendComplete): Ditto.
* Source/WebCore/platform/graphics/PlatformTimeRanges.cpp:
(WebCore::PlatformTimeRanges::toString const): Added String conversion method.
* Source/WebCore/platform/graphics/PlatformTimeRanges.h:
(WTF::LogArgument<PlatformTimeRanges>::toString): Ditto.

Canonical link: https://commits.webkit.org/256317@main
